lib/wallet/src/Cardano/Wallet/Write/ProtocolParameters.hs

```diff
@@ -23,28 +22,25 @@ import qualified Cardano.Wallet.Write.Tx as WriteTx
 -- TODO:
 --  - Make this data type abstract: don't export the constructor.
 --  - Replace this type with a re-exported 'Ledger.PParams era'.
-data ProtocolParameters era = ProtocolParameters
-    { pparamsWallet
-        :: !Wallet.ProtocolParameters
-    , pparamsLedger
-        :: !(WriteTx.PParams (WriteTx.ShelleyLedgerEra era))
+newtype ProtocolParameters era = ProtocolParameters
+    { pparamsLedger
+        :: (WriteTx.PParams (WriteTx.ShelleyLedgerEra era))
     }
 
 -- TODO: ADP-2459 - replace with something nicer.
 unsafeFromWalletProtocolParameters
     :: forall era. CardanoApi.IsShelleyBasedEra era
     => Wallet.ProtocolParameters
     -> ProtocolParameters era
-unsafeFromWalletProtocolParameters pparamsWallet =
-    ProtocolParameters {pparamsWallet, pparamsLedger}
-  where
-    pparamsLedger = maybe
+unsafeFromWalletProtocolParameters pparamsWallet = ProtocolParameters $
+    maybe
         (error missingNodeParamsError)
         (CardanoApi.unbundleLedgerShelleyBasedProtocolParams
             (CardanoApi.shelleyBasedEra @era)
             . CardanoApi.bundleProtocolParams (CardanoApi.cardanoEra @era)
```

Comment:
I find these multi-line expressions (in parentheses) a bit hard to read.

Perhaps we could write:
```suggestion
        unbundleParameters
```

And in the `where` clause, add the following definition:
```hs
        unbundleParameters
            = CardanoApi.unbundleLedgerShelleyBasedProtocolParams
                (CardanoApi.shelleyBasedEra @era)
            . CardanoApi.bundleProtocolParams
                (CardanoApi.cardanoEra @era)
```

Reply:
done